File: 卡码/代码随想录/1/09-动态规划/Go/05-子序列问题/02-子序列(连续)/LeetCode_0674.go
// 674. 最长连续递增序列
package main

func findLengthOfLCIS(nums []int) int {
	dp := make([]int, len(nums))
	for i := 0; i < len(dp); i++ {
		dp[i] = 1
	}
	res := 1
	for i := 1; i < len(nums); i++ {
		if nums[i] > nums[i-1] {
			dp[i] = dp[i-1] + 1
		}
		if dp[i] > res {
			res = dp[i]
		}
	}
	return res
}

File: 卡码/代码随想录/1/09-动态规划/Java/dp/src/main/java/com/kurumi/dp/_05_子序列问题/_02_连续子序列/LeetCode_0674.java
//674. 最长连续递增序列
package com.kurumi.dp._05_子序列问题._02_连续子序列;

import java.util.Arrays;

public class LeetCode_0674 {

}
class Solution {
  public int findLengthOfLCIS(int[] nums) {
    int[] dp=new int[nums.length];
    Arrays.fill(dp,1);
    int res=1;
    for(int i=1;i<nums.length;i++){
      if(nums[i]>nums[i-1]) dp[i]=dp[i-1]+1;
      res=res>dp[i]?res:dp[i];
    }
    return res;
  }
}
